src/esp/gfx/Simulator.h

```diff
@@ -5,14 +5,14 @@
 #pragma once
 
 #include "WindowlessContext.h"
-#include "esp/agent/Agent.h"
-#include "esp/assets/ResourceManager.h"
 #include "esp/core/esp.h"
 #include "esp/core/random.h"
 #include "esp/scene/SceneConfiguration.h"
 #include "esp/scene/SceneManager.h"
 #include "esp/scene/SceneNode.h"
 
+#include "esp/assets/ResourceManager.h"
```

Comment:
Is there a reason for re-ordering this header include to be last?

Reply:
There is a name collisions somewhere if that header is include further up.  Previously it seems that having the header for the agent included dealt with whatever the issue is (the header for the agent was before that header)